src/wallet/Web3WalletProvider.ts

```diff
@@ -97,15 +97,20 @@ export class Web3WalletProvider {
 		if (!walletType) throw new Error('Please provide a Wallet type to connect with.')
 
 		if (this[walletType as keyof Web3WalletProvider]) {
-			// @ts-ignore
-			// TODO: this address is null for the flow network
-			// Actual connected address is get in flowSubscribe function
-			const address = await this[walletType as keyof Web3WalletProvider](checkConnectionOnly)
-			logger(2, 'address', address)
+			try {
+				// @ts-ignore
+				// TODO: this address is null for the flow network
+				// Actual connected address is get in flowSubscribe function
+				let address = await this[walletType as keyof Web3WalletProvider](checkConnectionOnly)
+				logger(2, 'address', address)
 
-			this.saveConnections()
-			this.emitSavedConnection(address)
-			return address
+				this.saveConnections()
+				this.emitSavedConnection(address)
+				return address
+			} catch (e) {
+				logger(2, 'Provider connect Error: ', e)
+				throw new Error(e.message)
```

Comment:
Let's remove this try/catch, or a the very least use "throw e;" rather than "throw new Error(e.message)" to retain the original stacktrace.

Reply:
done